terraform/context_apply_test.go

```diff
@@ -10521,7 +10521,6 @@ func TestContext2Apply_plannedDestroyInterpolatedCount(t *testing.T) {
 	}
 
 	ctxOpts.ProviderResolver = providerResolver
-	ctxOpts.Destroy = true
```

Comment:
I'm a little confused by this particular test because it's in the `context_apply_test` file and its name says `TestContext2Apply_` but it seems to depend on a `plan-`-prefixed fixture. :thinking: 

Given the comment above it does seem like the goal here was to emulate a separate `terraform plan -destroy -out=tfplan` then `terraform apply tfplan`, in which case this removal does make this more realistic. It might be helpful to have a comment here explicitly noting that we're trying to test a saved destroy plan and so we're intentionally not setting `ctxOpts.Destroy` for that reason, to match the real-world behavior of that scenario.

Reply:
Yes, since this test is specifically checking against the behavior of a saved plan, it should never have had the `Destroy` opt set. Since that was added as part of the "big ugly change", I can only assume it was there to make the test pass at the time.